addons/sale_stock/models/sale_order.py

```diff
@@ -13,7 +13,7 @@ class SaleOrder(models.Model):
 
     @api.model
     def _default_warehouse_id(self):
-        company = self.env.user.company_id.id
+        company = self.env.context.get('company_id') or self.env.user.company_id.id
```

Comment:
```suggestion
        company_id = int(self.default_get(['company_id']).get('company_id')) or self.env.user.company_id.id
        return self.env['stock.warehouse'].search([('company_id', '=', company_id)], limit=1)
```
Please use a clean `default_warehouse_id` context key instead of this confusing `company_id` default.

Reply:
ups, I didn't see this comment in time @Feyensv , sorry. Well, I updated the branch with your suggestion (it's not reflected because the PR is closed). If you open it again the PR, then you could merge it.